### PR TITLE
Documentation for auth handler

### PR DESCRIPTION
**Pull request by sfan5**
+a small change to discourage modders from unportable practices

please <b>don't squash</b> during merge

### CODE REVIEW

File: doc/lua_api.txt [hunk 1]
```diff
@@ -2654,37 +2665,44 @@ Call these functions only at load time!
...
 	* `player_or_name`: Either a Player object or the name of a player.
 	* `...` is either a list of strings, e.g. `"priva", "privb"` or
 	  a table, e.g. `{ priva = true, privb = true }`.
-* `minetest.get_player_ip(name)`: returns an IP address string
+
+* `minetest.check_password_entry(name, entry, password)`
+    * Returns true if the "password entry" for a player with name matches given
+      password, false otherwise.
+    * The "db entry" is the opaque password representation generated by the engine,
+      as returned by the `get_auth()` method of the auth handler
```
Comment (SmallJoker) [inline doc/lua_api.txt]: What "db entry"?

File: doc/lua_api.txt [hunk 2]
```diff
@@ -2654,37 +2665,44 @@ Call these functions only at load time!
...
+      in the db might use the new SRP verifier format.
+    * For this purpose, use `minetest.check_password_entry` instead.
+* `minetest.notify_authentication_modified(name)`
+    * Should be called by the authentication handler if privileges changes.
+    * `name`: string, if omitted, all auth data should be considered modified
+* `minetest.get_player_ip(name)`: returns an IP address string for the player `name`
+    * The player needs to be online for this to be successful.
+
+* `minetest.get_auth_handler()`: Return the currently active auth handler
+    * See authentication handler definition
```
Comment (SmallJoker) [inline doc/lua_api.txt]: This `authentication handler definition` reference is not obvious where it leads to.

File: doc/lua_api.txt [hunk 3]
```diff
@@ -2654,37 +2665,44 @@ Call these functions only at load time!
...
+      external protocols such as IRC, other uses are frowned upon.
+* `minetest.get_password_hash(name, raw_password)`
+    * Convert a name-password pair to a password hash that Minetest can use.
+    * The returned value alone is not a good basis for password checks based
+      on comparing the password hash in the database with the password hash
+      from the function, with an externally provided password, as the hash
+      in the db might use the new SRP verifier format.
+    * For this purpose, use `minetest.check_password_entry` instead.
+* `minetest.notify_authentication_modified(name)`
+    * Should be called by the authentication handler if privileges changes.
```
Comment (SmallJoker) [inline doc/lua_api.txt]: Plus something like: `Sends the affected player(s) their new privilege list`

File: doc/lua_api.txt (hunk 1 above)
Comment (sfan5) [reply doc/lua_api.txt]: copied from old description

File: doc/lua_api.txt (hunk 3 above)
Comment (sfan5) [reply doc/lua_api.txt]: Implementation detail, I wouldn't add that to the docs.

Comment (SmallJoker) [reply doc/lua_api.txt]: IMO, this makes an important difference to let people know what this function does.

Comment (sfan5) [reply doc/lua_api.txt]: It really doesn't no, this isn't even anything user visible.

**Comment by BluebirdGreycoat** [comment]
Thanks for breaking my server code, heh. I needed that to scan user names, to ensure that no one can choose a temporary nick that is the same as or similar to someone elses.

**Comment by rubenwardy** [comment]
> Thanks for breaking my server code, heh. I needed that to scan user names, to ensure that no one can choose a temporary nick that is the same as or similar to someone elses.

It's an undocumented API, so no guarantees it will be kept. Would be good to have some way to search through usernames though

**Comment by BluebirdGreycoat** [comment]
To be precise, I need to iterate the auth table in order to compare a lowercase version of each entry with a lowercase version of another name, and do something if a match is found. How can I do this without completely re-implementing the whole auth system?

**Comment by BluebirdGreycoat** [comment]
Proposal: `minetest.auth_name_compare(playername, casesensitive)`

Function should iterate through the auth table and compare `playername` with each entry, optionally doing so case-sensitively if `casesensitive` is `true`. Function should return the number of matches found, or 0 if no match was found.

Thoughts?

**Comment by BluebirdGreycoat** [comment]
#6700

**Pull request merged**